src/org/opendatakit/validate/FormValidator.java

```diff
@@ -331,14 +341,37 @@ boolean stepThroughEntireForm(FormEntryModel model) throws InvalidReferenceExcep
     }
 
     public void validateAndExitWithErrorCode(String path) {
-        try {
-            validate(path);
-        } catch (Exception e) {
-            errors.error("\nException: ", e);
-            setError(true);
+        validateAndExitWithErrorCode(new String[] { path }, true);
+    }
+
+    public void validateAndExitWithErrorCode(String[] paths, boolean failFast) {
```

Comment:
Why not accept a `List<String>` instead of a `String[]`? That way you can avoid the ugly casting from array to list in the calling site.

Reply:
There's not really a strong reason, if you'd prefer the arguments to be a `List<String>` I can change it.

Reply:
I don't have strong feelings about it. It's just that the inly source for that argument is a `List`. It feels more straightforward to use it as is, instead of casting to arrays.

Reply:
Done